eZ/Publish/Core/FieldType/Validator/DestinationContentValidator.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\FieldType\Validator;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\Exceptions\UnauthorizedException;
+use eZ\Publish\Core\FieldType\ValidationError;
+
+/**
+ * Validator for checking existence of destination content.
+ */
+class DestinationContentValidator
+{
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
+
+    public function __construct(?ContentService $contentService = null)
+    {
+        $this->contentService = $contentService;
+    }
+
+    public function validate($value): ?ValidationError
+    {
+        try {
+            $this->contentService->loadContentInfo($value);
```

Comment:
The validator only checks if object can be loaded, it doesn't check of the the objects content type is allowed or not by the `selectionContentTypes` setting

Reply:
Resolved in https://github.com/ezsystems/ezpublish-kernel/pull/3131/commits/2e54900ef4201a45805af00bdec643c2dad25187